server/test/unit/server/webservices/views/test_util.py

```diff
@@ -166,6 +187,63 @@ def mock_function(self, request):
         self.assertEqual(response.http_status_code, httplib.BAD_REQUEST)
         self.assertEqual(response.error_code.code, 'PLP1009')
 
+    def test_json_type_right(self):
+        """
+        Test that valid JSON is unchanged if it is of correct type.
+        """
+        mock_request = mock.MagicMock()
+        mock_expected_json = {"expected": "datatype"}
+        mock_request.body = json.dumps(mock_expected_json)
+
+        @parse_json_body(json_type=dict)
+        def mock_function(self, request):
+            return request
+
+        self.assertEqual(mock_function(self, mock_request).body_as_json, mock_expected_json)
+
+    def test_json_type_wrong(self):
+        """
+        Test that incorrect type of JSON results in a 400.
+        """
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps(["unexpected datatype"])
+
+        @parse_json_body(json_type=dict)
+        def mock_function(self, request):
+            return request
+
+        with self.assertRaises(PulpCodedValidationException) as cm:
```

Comment:
This failed in the jenkins run because on python 2.6, it doesn't have the context manager capability. We handle that sort of problem by using ``unittest2``, which is a backport of the fancy new test tools to python 2.6. You can do this to get the "right" one for your platform:

``from pulp.common.compat import unittest``